src/base/bittorrent/peerinfo.cpp

```diff
@@ -367,6 +367,10 @@ void PeerInfo::determineFlags()
     if (useUTPSocket())
         updateFlags(u'P', C_UTP);
 
+    // h = Peer is using UDP hole punching
+    if (isHolepunched())
+        updateFlags(u'h', tr("Peer was found using a NAT hole punch"));
```

Comment:
Maybe "Connected via UDP hole-punching"?

Reply:
A peer with this flag can merely be found, but not necessarily connected like DHT or PEX.
![2024-07-11_204901](https://github.com/qbittorrent/qBittorrent/assets/63958081/66d62b99-bff8-46da-877c-3ab7e155ac69)

Reply:
Well i'm lazy to find out what hole punch is, so the best i can offer is to fix the current string:
```suggestion
        updateFlags(u'h', tr("Peer was found by using a NAT hole punch"));
```

Reply:
How about even shorter?? =
Peer was using a NAT hole punch

Btw, should it be in past tense?

Reply:
> UDP hole punching is a commonly used technique employed in network address (NAT) applications

> UDP hole punching establishes connectivity between two hosts communicating across one or more network address translators.

Why "found"? Kind of sounds like a peer uses it to connect to other peers in peer-to-peer applications because of the peers private network setup.

Could this be used instead?? =
Peer is using NAT hole punching
or
Peer is using UDP hole punching

Reply:
> Why "found"? Kind of sounds like a peer uses it to connect to other peers in peer-to-peer applications because of the peers LAN setup.

As shown in the screenshot, peers with this flag are not connected, but only trying to connect. They are just present in the list of peers. Using “Connected” will be incorrect.
"Peer was found using a NAT hole punch" it's just an expanded description from this one "Peer is using NAT hole punching"

Reply:
For consistency with the other descriptions: "Peer using NAT hole punch"

Reply:
>For consistency with the other descriptions: "Peer using NAT hole punch"

This looked the best to me.

@stalkerok Do you mind changing to it?

Reply:
> For consistency with the other descriptions: "Peer using NAT hole punch"

Actually, there are no similar descriptions, there are similar comments, and in the comments, I did a similar thing. Ok, I've changed the description.